### DISCUSSION TITLE
Used ports of player providers in documentation

### DISCUSSION THREAD

**Discussion by arboeh** [music-assistant/support#1965] [Feature requests and ideas]
For me it's always hard to find the ports of the different protocols and so it's hard for me to create the correct firewall rules.
Is it possible to note it in the respective music provider documentation (https://music-assistant.io/player-support/)?
Right now I have problems to find the correct ports for DLNA/UPnP player provider. Still doesn't work with 1900 udp and 8200 tcp...

**Comment by OzGav** [comment]
Edit: I see you are trying to write firewall rules. We don’t recommend trying to run MA behind a firewall

**Reply by arboeh** [reply]
I'm using Home Assistant with Music Assistant as plugin on the same host and I'm able to use Home Assistant from outside my network over the Internet with Cloudflared add-on.
I'm not a system admin and I hate the port and ip things, but if I understand it right, the system should have to be behind the firewall and so I need the ports to create firewall rules, or not?

**Reply by OzGav** [reply]
If all your players are on the same network as HA/MA then no you don’t. I can access my system from outside the network and control all the players . You can’t hear anything because it is playing at home but that is a direct connection between MA and the player, nothing needs to leave the network and thus there are no firewall rules required

**Reply by OzGav** [reply]
The above is written assuming you have successfully connected to HA from outside your network

**Reply by arboeh** [reply]
Sorry, I didn't mention that my player is an Raspi, that is in the same network, connected over WIFI. The player isn't on the same host, where HA/MA is running, so there is no "direct connection".

**Reply by OzGav** [reply]
That doesn’t matter. I have Chromecasts, RPi running picoreplayer, AirPlay devices, Fully Kiosk tablets and I can initiate playback to all of them from anywhere

**Reply by OzGav** [reply]
The direct connection I am talking about is the connection between MA and the player. You interacting with MA from outside your network has no effect on that direct connection

**Reply by arboeh** [reply]
Okay, thanks. I will take a look again.
